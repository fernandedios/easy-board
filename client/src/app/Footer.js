import React from 'react';

const Footer = () => {
    return (
        <footer className="footer">
            <div className="container">
                <div className="content has-text-centered">
                    <p>&copy; Easy Board. Open Source under MIT License.</p>
                </div>
            </div>
        </footer>
    );
};

export default Footer;
