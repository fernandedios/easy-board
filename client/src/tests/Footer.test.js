import React from 'react';
import { shallow, mount } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';

import Footer from '../app/Footer';

describe('<Footer />', () => {
    it('should be defined', () => {
        expect(Footer).toBeDefined();
    });
    
    it('should have correct classNames', () => {
        const output = shallow(<Footer />);
        expect(output.find('footer').hasClass('footer')).toEqual(true);
    });

    it('should render footer notice', () => {
        const output = mount(<Footer />);
        expect(output.find('.content').text()).toEqual('Â© Easy Board. Open Source under MIT License.');
    });

    it('should render correctly', () => {
        const output = ReactTestRenderer.create(<Footer />).toJSON();
        expect(output).toMatchSnapshot();
    });
});
