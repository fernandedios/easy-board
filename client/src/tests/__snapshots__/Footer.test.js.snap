// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="content has-text-centered"
    >
      <p>
        Â© Easy Board. Open Source under MIT License.
      </p>
    </div>
  </div>
</footer>
`;
